### ISSUE TITLE
System.InvalidOperationException: variable ... of type AgileObjects.AgileMapper.ObjectPopulation.IObjectMappingData ... but is not defined

### ISSUE THREAD

**Issue by bslbckr** [agileobjects/AgileMapper#174]
Using the freshly provided 1.6.0-preview4 version of the mapper I ran into the above mentioned exception using a relatively simple object structure. I tried to map the following classes:
```csharp
namespace Src {
	public class Definition {
		public enum Scope {
			GLOBAL, LANE, INDEX			
		}
		public string Identification {get;set;}
		public string Unit {get;set;}
		public IList<int> UnavailableNumbers {get;set;}
		public Scope DefScope {get;set;}
		public object Min {get;set;}
		public object Max {get;set;}
	}
}

namespace Tgt {
  public class ParamDef {
	  public enum Scope {
		  STH, COMPLETELY, DIFFERENT
	  }
	  public string Identification {get;set;}
	  public string Unit {get;set;}
	  public IList<int> UnavailableNumbers{get;set;}
	  public Scope DefScope {get;set;}
	  public Wrapper Min {get;set;}
	  public Wrapper Max {get;set;}
  }
	
	public class Wrapper {
		public int IntVal {get;set;}
		public bool BoolVal {get;set;}
		public string StringVal {get;set;}
		public double DoubleVal {get;set;}
	}
}
```
from namespace `Src` to namespace `Tgt` using the following mapping-specification:
```csharp
Mapper.WhenMapping.PairEnums<Src.Definition.Scope>(new [] {Src.Definition.Scope.GLOBAL, Src.Definition.Scope.LANE, Src.Definition.Scope.INDEX}).With<Tgt.ParamDef.Scope>(new [] {Tgt.ParamDef.Scope.STH, Tgt.ParamDef.Scope.COMPLETELY, Tgt.ParamDef.Scope.DIFFERENT});
Mapper.WhenMapping.From<int>().To<Tgt.Wrapper>().Map(input => input, tgt => tgt.IntVal);
```
If I omit the `Min` and `Max` members, the mapping works flawlessly. However, if I include them I receive the following exception:
```
[System.InvalidOperationException: variable 'dToPdData' of type 'AgileObjects.AgileMapper.ObjectPopulation.IObjectMappingData`2[Src.Definition,Tgt.ParamDef]' referenced from scope '', but it is not defined]
   at System.Linq.Expressions.Compiler.VariableBinder.Reference(ParameterExpression node, VariableStorageKind storage)
   at System.Linq.Expressions.Compiler.VariableBinder.VisitParameter(ParameterExpression node)
   at System.Linq.Expressions.ParameterExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitMember(MemberExpression node)
   at System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitMember(MemberExpression node)
   at System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitMethodCall(MethodCallExpression node)
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitArguments(IArgumentProvider nodes)
   at System.Linq.Expressions.ExpressionVisitor.VisitMethodCall(MethodCallExpression node)
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitConditional(ConditionalExpression node)
   at System.Linq.Expressions.ConditionalExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.Visit(ReadOnlyCollection`1 nodes)
   at System.Linq.Expressions.Compiler.VariableBinder.VisitBlock(BlockExpression node)
   at System.Linq.Expressions.BlockExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitBinary(BinaryExpression node)
   at System.Linq.Expressions.BinaryExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.Visit(ReadOnlyCollection`1 nodes)
   at System.Linq.Expressions.Compiler.VariableBinder.VisitBlock(BlockExpression node)
   at System.Linq.Expressions.BlockExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitTry(TryExpression node)
   at System.Linq.Expressions.TryExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.Visit(ReadOnlyCollection`1 nodes)
   at System.Linq.Expressions.Compiler.VariableBinder.VisitLambda[T](Expression`1 node)
   at System.Linq.Expressions.Expression`1.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.Compiler.VariableBinder.Visit(Expression node)
   at System.Linq.Expressions.Compiler.LambdaCompiler.Compile(LambdaExpression lambda, DebugInfoGenerator debugInfoGenerator)
   at System.Linq.Expressions.Expression`1.Compile()
   at AgileObjects.AgileMapper.ObjectPopulation.ObjectMapper`2..ctor(Expression mapping, IObjectMappingData mappingData)
   at AgileObjects.AgileMapper.ObjectPopulation.ObjectMapperFactory.Create[TSource,TTarget](ObjectMappingData`2 mappingData)
   at AgileObjects.AgileMapper.ObjectPopulation.ObjectMappingData`2.GetOrCreateMapper()
   at AgileObjects.AgileMapper.ObjectPopulation.ObjectMapper`2.<>c.<MapSubObject>b__18_0(ObjectMapperKeyBase key)
   at AgileObjects.AgileMapper.Caching.ArrayCache`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at AgileObjects.AgileMapper.ObjectPopulation.ObjectMapper`2.MapSubObject(IObjectMappingData mappingData)
   at AgileObjects.AgileMapper.ObjectPopulation.ObjectMappingData`2.Map[TDeclaredSource,TDeclaredTarget](TDeclaredSource sourceValue, TDeclaredTarget targetValue, String targetMemberRegistrationName, Int32 dataSourceIndex)
   at lambda_method(Closure , IObjectMappingData`2 )

[AgileObjects.AgileMapper.MappingException: An exception occurred mapping Definition -> ParamDef with rule set CreateNew.]
   at lambda_method(Closure , IObjectMappingData`2 )
   at AgileObjects.AgileMapper.ObjectPopulation.ObjectMapper`2.Map(ObjectMappingData`2 mappingData)
   at AgileObjects.AgileMapper.MappingExecutor`1.PerformMapping[TTarget](TTarget target)
   at AgileObjects.AgileMapper.MappingExecutor`1.PerformMapping[TTarget](MappingRuleSet ruleSet, TTarget target)
   at AgileObjects.AgileMapper.MappingExecutor`1.ToANew[TResult]()
   at Program.Main() :line 50
```

The complete example could be found in this [fiddle](https://dotnetfiddle.net/T3KvE4)

**Comment by bslbckr** [comment]
Any news on this one?
Is there anything I could help with? e.g. writing a pr with an appropriate Unit-Test for this issue or sth similar?

**Comment by SteveWilkes** [comment]
Hi!

Sorry for the delay - I've not got around to looking at this yet. I will, and will report back.

Cheers,

Steve

**Comment by SteveWilkes** [comment]
This is fixed in the latest v1.6 branch code - release to follow! Cheers! :)

**Comment by SteveWilkes** [comment]
This fix is included in v1.6-preview5, now available [on NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper). Thanks again!

**Comment by SteveWilkes** [comment]
The fix for this is included in v1.6, which is now available [on NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper/1.6.0).

Thanks again for the feedback!
